### PR TITLE
[TensorExpr] Implement 'call_raw' in IREval.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57901 [TensorExpr] Implement 'call_raw' in CUDA codegen.
* **#57882 [TensorExpr] Implement 'call_raw' in IREval.**
* #57881 [TensorExpr] Minor cleanup in IREval.

Differential Revision: [D28306752](https://our.internmc.facebook.com/intern/diff/D28306752)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

File: torch/csrc/jit/tensorexpr/eval.cpp [hunk 1]
```diff
@@ -985,22 +993,20 @@ void SimpleIREvaluator::call(const std::vector<CallArg>& args) {
...
 
   switch (bufArg.dtype().scalar_type()) {
-#define TYPE_CASE(Type, Name)                        \
-  case ScalarType::Name:                             \
-    impl_->bindVar(bufArg.var(), *data.Name##Ptr()); \
-    break;
+#define TYPE_CASE(Type, Name)                 \
+  case ScalarType::Name: {                    \
+    Type typed_data;                          \
+    memcpy(&typed_data, &data, sizeof(Type)); \
```
Comment (navahgar) [inline torch/csrc/jit/tensorexpr/eval.cpp:1006]: We seem to be adding an extra `memcpy` here. Isn't this increasing the overhead? Should we retain the old code for when it is called using `CallArg`?

**Review by navahgar — APPROVED** [review]
LGTM

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28306752).

**Pull request closed without merging**